AutomatedTesting/Gem/PythonTests/prefab/Prefab_Operations.py

```diff
@@ -0,0 +1,190 @@
+        "target entity does *not* owns given count of children")
+
+    delete_prefab_succeeded = (
+        "'DeleteEntitiesAndAllDescendantsInInstance' passed",
+        "'DeleteEntitiesAndAllDescendantsInInstance' failed")
+
+    delete_prefab_entities_and_descendants_succeeded = (
+        "Entities and all the descendants in target prefab are deleted",
+        "Entities and all the descendants in target prefab are *not* deleted")
+
+    reparent_prefab_no_cyclical_dependency_detected = (
+        "Cyclical dependency not detected while reparenting prefab",
+        "Cyclical dependency detected while reparenting prefab")
+
+    reparent_prefab_new_child_with_reparented_prefab_name_added = (
+        "An entity with reparented prefab name becomes a child of target parent entity",
+        "No entity with reparented prefab name become a child of target parent entity")
+
+    reparent_prefab_reparented_prefab_under_parent = (
+        "Prefab reparented is under the expected parent entity",
+        "Prefab reparented is *not* under the expected parent entity")
+
+
+def create_entity(entity_name=None, parent_entity_id=EntityId()):
+    new_entity_id = editor.ToolsApplicationRequestBus(bus.Broadcast, 'CreateNewEntity', parent_entity_id)
+    Report.result(OperationResults.create_entity_succeeded, new_entity_id.IsValid())
+    
+    if new_entity_id.IsValid():
+        if entity_name is not None:
+            editor.EditorEntityAPIBus(bus.Event, 'SetName', new_entity_id, entity_name)
+        return new_entity_id
+    else:
+        return EntityId()
+
+
+def create_prefab(entity_ids, file_name, prefab_name=None):
+    file_path = prefab_test_utils.get_prefab_file_path(file_name)
+    create_prefab_result = prefab.PrefabPublicRequestBus(bus.Broadcast, 'CreatePrefabInMemory', entity_ids, file_path)
+    Report.result(OperationResults.create_entity_succeeded, create_prefab_result.IsSuccess())
+    
+    if not create_prefab_result.IsSuccess():
+        return EntityId()
+
+    container_entity_id = create_prefab_result.GetValue()
+
+    if prefab_name is not None:
+        editor.EditorEntityAPIBus(bus.Event, 'SetName', container_entity_id, prefab_name)
+    else:
+        prefab_name = file_name
+
+    prefab_test_utils.wait_for_propagation()
+    container_entity_id = prefab_test_utils.find_entity_by_unique_name(prefab_name)
+    return container_entity_id
+
+
+def instantiate_prefab(file_name, prefab_name=None, parent_entity_id=EntityId(), expected_prefab_position=Vector3()):
+    file_path = prefab_test_utils.get_prefab_file_path(file_name)
+    instantiate_prefab_result = prefab.PrefabPublicRequestBus(
+        bus.Broadcast, 'InstantiatePrefab', file_path, parent_entity_id, expected_prefab_position)
+    Report.result(OperationResults.instantiate_prefab_succeeded, instantiate_prefab_result.IsSuccess())
+
+    if not instantiate_prefab_result.IsSuccess():
+        return EntityId()
```

Comment:
you could return None here instead

Reply:
I would prefer using critical_result here instead of using None. That way I won't have to check for None later.

Reply:
Use critical_result here for the fix.